design_docs/protostar-architecture.md

```diff
@@ -0,0 +1,63 @@
+# Protostar modular architecture
+
+This document proposes the idea how to modularize Protostar. Goals of the modularization are
+- Allow to introduce better ownerships of the codebase parts
+- Make workflow asynchronous between smaller sub-teams operating on contracts between modules
+- Allow incremental rewrite to Rust
+- Remove compilation responsibility from Protostar codebase
+
+
+
+## General overview
+First decomposition of Protostar will introduce 3 modules
```

Comment:
What about CLI itself? I guess "gateway" will have this module inside (parsing args, etc) right?

Reply:
Looks like so

Reply:
@Arcticae yup, communication contract with Scarb will be the reposnsibility of CLI